File: app/src/main/java/info/nightscout/androidaps/plugins/PumpInsightLocal/app_layer/remote_control/CancelBolusMessage.java
package info.nightscout.androidaps.plugins.PumpInsightLocal.app_layer.remote_control;

import info.nightscout.androidaps.plugins.PumpInsightLocal.app_layer.AppLayerMessage;
import info.nightscout.androidaps.plugins.PumpInsightLocal.app_layer.Service;
import info.nightscout.androidaps.plugins.PumpInsightLocal.descriptors.MessagePriority;
import info.nightscout.androidaps.plugins.PumpInsightLocal.utils.ByteBuf;

public class CancelBolusMessage extends AppLayerMessage {

    private int bolusID;

    public CancelBolusMessage() {
        super(MessagePriority.HIGHEST, false, true, Service.REMOTE_CONTROL);
    }

    @Override
    protected ByteBuf getData() {
        ByteBuf byteBuf = new ByteBuf(2);
        byteBuf.putUInt16LE(bolusID);
        return byteBuf;
    }

    public void setBolusID(int bolusID) {
        this.bolusID = bolusID;
    }
}

File: pump/insight/src/main/kotlin/app/aaps/pump/insight/app_layer/remote_control/CancelBolusMessage.kt
package app.aaps.pump.insight.app_layer.remote_control

import app.aaps.pump.insight.app_layer.AppLayerMessage
import app.aaps.pump.insight.app_layer.Service
import app.aaps.pump.insight.descriptors.MessagePriority
import app.aaps.pump.insight.utils.ByteBuf

class CancelBolusMessage : AppLayerMessage(MessagePriority.HIGHEST, false, true, Service.REMOTE_CONTROL) {

    internal var bolusID = 0
    override val data: ByteBuf
        get() {
            val byteBuf = ByteBuf(2).apply { putUInt16LE(bolusID) }
            return byteBuf
        }
}